src/Data/Singletons/Partition.hs

```diff
@@ -224,6 +235,18 @@ partitionDeriving mb_strat deriv_pred mb_ctxt ty data_decl =
        , deriv_name == enumName
       -> mk_derived_inst <$> mk_instance mkEnumInstance
 
+       | stock_or_default
```

Comment:
This region could be cleaned up a bit by having an mapping (or an associative list, at list) from class names to instance makers.

Reply:
Good idea, I'll do that.